File: page-objects/js-mocha-chai/complete/pages/bag-page.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

// var SearchResultsPage = require('./search-results-page.js');

BagPage = function BagPage(driver) {
    this.driver = driver;
    this.articleName = By.name("cart.product.name");
};

BagPage.prototype.getArticleName = function() {
    return this.driver.findElement(this.articleName).getText();
};


module.exports = BagPage;

File: page-objects/java-fluentlenium/complete/src/main/java/pages/BagPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class BagPage {

    private WebDriver webDriver;

    @FindBy(name = "cart.product.name")
    private WebElement articleName;

    public BagPage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public String getArticleName() {
        return articleName.getText();
    }
}
